### PR TITLE
feat: validate data column sidecars

### PR DESCRIPTION
**Pull request by matthewkeil**
**Motivation**

Implements `validateDataColumnsSidecars` in place of stub function

### CODE REVIEW

File: packages/beacon-node/src/chain/validation/dataColumnSidecar.ts [hunk 1]
```diff
@@ -45,14 +47,60 @@ export async function validateGossipDataColumnSidecar(
...
+      );
+    }
+
+    if (columnIndex < NUMBER_OF_COLUMNS) {
+      throw new Error(
+        `Invalid data column columnIndex=${columnIndex} in slot=${blockSlot} blockRoot=${toHex(blockRoot)} sidecarsIndex=${sidecarsIndex}`
+      );
+    }
+
+    const {column, kzgCommitments, kzgProofs} = columnSidecar;
```
Comment (g11tech) [inline packages/beacon-node/src/chain/validation/dataColumnSidecar.ts]: you need to match these commitments with the expected commitments from the block

File: packages/beacon-node/src/chain/validation/dataColumnSidecar.ts [hunk 2]
```diff
@@ -45,14 +47,60 @@ export async function validateGossipDataColumnSidecar(
 }
 
 export function validateDataColumnsSidecars(
```
Comment (g11tech) [inline packages/beacon-node/src/chain/validation/dataColumnSidecar.ts:49]: add a unit test?

File: packages/beacon-node/src/chain/validation/dataColumnSidecar.ts (hunk 1 above)
Comment (matthewkeil) [reply packages/beacon-node/src/chain/validation/dataColumnSidecar.ts]: good catch.  updated!

File: packages/beacon-node/src/chain/validation/dataColumnSidecar.ts (hunk 2 above)
Comment (matthewkeil) [reply packages/beacon-node/src/chain/validation/dataColumnSidecar.ts:49]: ok! added

**Review by g11tech — APPROVED** [review]

**Pull request merged**